app/components/candidate_interface/rejection_reasons_component.rb

```diff
@@ -61,8 +61,8 @@ def rejection_reasons_row(application_choice)
 
     def rejected_application_choices
       @rejected_application_choices ||= begin
-        rejected_applications = @application_form.application_choices.includes(:course, :provider, :current_course_option, :current_course).rejected
-        rejected_applications = rejected_applications.where('application_choices.rejection_reason IS NOT NULL OR application_choices.structured_rejection_reasons IS NOT NULL')
+        rejected_applications = @application_form.application_choices.includes(:course, :provider, :current_course_option, :current_course).rejected.or(@application_form.application_choices.offer_withdrawn)
+        rejected_applications = rejected_applications.where('application_choices.rejection_reason IS NOT NULL OR application_choices.structured_rejection_reasons IS NOT NULL OR application_choices.offer_withdrawal_reason IS NOT NULL')
```

Comment:
It's an odd mix of ActiveRecord's `.or` and native SQL here.
I'd suggest if it can be done using the AR DSL it should be done that way even if it's a bit longer in terms of code.

Also the nature of this component has changed, perhaps it should be renamed ie.  `RejectionAndWithdrawalReasonsComponent`.

Also there's a code smell here (not related to change made in this PR)
Is it the responsibility of a view tier component class to be retrieving these records? Shouldn't they be passed to the component? It seems a neater fit to have `RejectionReasonsComponent` accept a collection of rejection reasons.

Reply:
I refactor to use the AR DSL. I'd like to rename but this class is used in so many places that I prefer to move to the tech debt card. I create the card for the rename and the code smell here: https://trello.com/c/qplu18Cy/715-rename-rejection-reasons-component